build/source/engine/derivforce.f90

```diff
@@ -52,125 +51,95 @@ subroutine derivforce(err,message)
- newSnowDenAdd      => mpar_data%var(iLookPARAM%newSnowDenAdd)      ! Pahaut 1976, additive factor for new snow density (kg m-3)
- newSnowDenMultTemp => mpar_data%var(iLookPARAM%newSnowDenMultTemp) ! Pahaut 1976, multiplier for new snow density applied to air temperature (kg m-3 K-1)
- newSnowDenMultWind => mpar_data%var(iLookPARAM%newSnowDenMultWind) ! Pahaut 1976, multiplier for new snow density applied to wind speed (kg m-7/2 s-1/2)
- newSnowDenMultAnd  => mpar_data%var(iLookPARAM%newSnowDenMultAnd)  ! Anderson 1976, multiplier for new snow density for Anderson function (K-1)
- newSnowDenBase     => mpar_data%var(iLookPARAM%newSnowDenBase)     ! Anderson 1976, base value that is rasied to the (3/2) power (K)
- ! assign pointers to radiation geometry variables
- im        => time_data%var(iLookTIME%im)                           ! month
- id        => time_data%var(iLookTIME%id)                           ! day
- ih        => time_data%var(iLookTIME%ih)                           ! hour
- imin      => time_data%var(iLookTIME%imin)                         ! minute
- latitude  => attr_data%var(iLookATTR%latitude)                     ! latitude (degrees north
- cosZenith => mvar_data%var(iLookMVAR%scalarCosZenith)%dat(1)       ! average cosine of the zenith angle over time step DT
- ! assign pointers to model forcing data
- SWRadAtm => forc_data%var(iLookFORCE%SWRadAtm)                     ! downward shortwave radiation (W m-2)
- airtemp  => forc_data%var(iLookFORCE%airtemp)                      ! air temperature at 2 meter height (K)
- windspd  => forc_data%var(iLookFORCE%windspd)                      ! wind speed at 10 meter height (m s-1)
- airpres  => forc_data%var(iLookFORCE%airpres)                      ! air pressure at 2 meter height (Pa)
- spechum  => forc_data%var(iLookFORCE%spechum)                      ! specific humidity at 2 meter height (g g-1)
- pptrate  => forc_data%var(iLookFORCE%pptrate)                      ! precipitation rate (kg m-2 s-1)
- ! assign pointers to derived model forcing data
- scalarO2air  => mvar_data%var(iLookMVAR%scalarO2air)%dat(1)        ! atmospheric o2 concentration (Pa)
- scalarCO2air => mvar_data%var(iLookMVAR%scalarCO2air)%dat(1)       ! atmospheric co2 concentration (Pa)
- ! assign pointers to radiation variables
- scalarFractionDirect    => mvar_data%var(iLookMVAR%scalarFractionDirect)%dat(1)    ! fraction of direct radiation (0-1)
- spectralIncomingDirect  => mvar_data%var(iLookMVAR%spectralIncomingDirect)%dat     ! downwelling direct shortwave radiation for each waveband (W m-2)
- spectralIncomingDiffuse => mvar_data%var(iLookMVAR%spectralIncomingDiffuse)%dat    ! downwelling diffuse shortwave radiation for each waveband (W m-2)
+ err=0; message="derivforce/"
+
+ ! check spectral dimension
  if(size(spectralIncomingDirect) /= 2 .or. size(spectralIncomingDiffuse) /= 2)then
   err=20; message=trim(message)//'expect two spectral classes for radiation'; return
  endif
```

Comment:
Yes, `nBands` would be cleaner here. The test is correct `/=` means "not equal to".